fastfold/distributed/comm.py

```diff
@@ -27,31 +27,48 @@ def _reduce(tensor: Tensor) -> Tensor:
     return tensor
 
 
-def _split(tensor: Tensor, dim: int = -1) -> Tensor:
+def _split(tensor: Tensor, dim: int = -1, drop_unused=False) -> Tensor:
     if gpc.get_world_size(ParallelMode.TENSOR) == 1:
         return tensor
 
     split_size = divide(tensor.shape[dim], gpc.get_world_size(ParallelMode.TENSOR))
     tensor_list = torch.split(tensor, split_size, dim=dim)
+    rank = gpc.get_local_rank(ParallelMode.TENSOR)
 
-    output = tensor_list[gpc.get_local_rank(ParallelMode.TENSOR)].contiguous()
+    if not drop_unused:
+        output = tensor_list[rank].contiguous()
+    else:
+        output = tensor_list[rank].contiguous().clone()
 
     return output
 
 
-def _gather(tensor: Tensor, dim: int = -1) -> Tensor:
+def _gather(tensor: Tensor, dim=-1, chunk_size=1) -> Tensor:
     if gpc.get_world_size(ParallelMode.TENSOR) == 1:
         return tensor
 
     if dim == 1 and list(tensor.shape)[0] == 1:
         output_shape = list(tensor.shape)
         output_shape[1] *= gpc.get_world_size(ParallelMode.TENSOR)
         output = torch.empty(output_shape, dtype=tensor.dtype, device=tensor.device)
-        tensor_list = output.chunk(gpc.get_world_size(ParallelMode.TENSOR), dim=1)
-        dist.all_gather(list(tensor_list),
-                        tensor,
-                        group=gpc.get_group(ParallelMode.TENSOR),
-                        async_op=False)
+
+        world_list = output.chunk(gpc.get_world_size(ParallelMode.TENSOR), dim=1)
+        tensor_list = []
+        for t in world_list:
+            tensor_list.extend(t.chunk(chunk_size, dim=1))
+
+        chunk_tensor = tensor.chunk(chunk_size, dim=1)
+
+        for i in range(chunk_size):
```

Comment:
Hi, the `gather` function will pass `chunk_size=None` by default to `_gather`.
And `range` can not process None, it should be processed manually.